src/main/java/UseCase/playPlaylist.java

```diff
@@ -1,4 +1,72 @@
 package UseCase;
 
-public class playPlaylist {
+import Entities.MusicPlayer;
+import Entities.Song;
+import InputBoundary.playPlaylistInputBoundary;
+import InputBoundary.playSongInputBoundary;
+import InputData.playlistInputData;
+import InputData.songInputData;
+import java.util.ArrayList;
+
+/**
+ * Use case layer class to play a playlist of songs. Handles the queuing of songs using synchronized
+ * methods in tandem with the MusicPlayer, but delegates playing individual songs to the play song
+ * use case. Thus, this use case has no output boundary, as the currently playing song is displayed
+ * by the play song use case.
+ */
+public class playPlaylist implements playPlaylistInputBoundary {
+    private boolean queue;
+    private final ArrayList<Song> playlist;
+    private final Object sync;
+    private int nextSong;
+
+    public playPlaylist(playlistInputData data){
+        this.playlist = data.getSongs();
+        queue = true;
+        nextSong = 0;
+        sync = MusicPlayer.getInstance().getSync();
+    }
+
+    /**
+     * Plays the playlist given in the constructor.
+     */
+    @Override
+    public void play() {
+        if (nextSong < playlist.size()){
+            playSongInputBoundary p = new playSong(new songInputData(playlist.get(nextSong)));
+            nextSong++;
+            final Thread t = new Thread(this::playNext);
+            t.start();
+            p.play();
+        }
+    }
+
+    /**
+     * Dequeues the playlist, such that after the currently playing song ends the playlist
+     * will not continue. Must be called whenever audio source is switched (eg. user plays new song,
+     * different playlist, or space).
```

Comment:
When someone clicks 'play' on a new song in the middle of a playPlaylist, is this insinuating that the current song must /finish/ playing before the song a user wanted plays?

Reply:
Nope! When someone calls play() again on the Music player, the current player is immediately killed. This method is here to prevent the _next_ song of the Playlist randomly interrupting the new content that was clicked on.